rise-storage.html

```diff
@@ -889,6 +945,81 @@
       });
 
       return response;
-    }
+    },
+
+    _loadStorage: function() {
+      if(!this.isOfflinePlayer()) {
+        this.$.storage.generateRequest();
+      }
+      else {
+        this.offlinePlayer.postMessage({
+          type: "storage-component-load",
+          clientId: this.offlinePlayerClientId,
+          url: this.$.storage.url }, "*");
+      }
+    },
+
+    /**
+     * Checks whether the page is running inside offline-player
+     */
+    isOfflinePlayer: function() {
+      return this.offlinePlayer;
+    },
+
+    /**
+     * Returns the path of the file/folder the component is listening to
+     */
+    getStoragePath: function() {
+      var path = "risemedialibrary-" + this.companyid + "/";
+
+      if(this.folder) {
+        path += this.folder + (this.folder.slice(-1) !== "/" ? "/" : "");
+      }
+
+      if(this.filename) {
+        path += this.filename;
+      }
+
+      return path;
+    },
+
+    /**
+     * Listens for events sent by offline player
+     */
+    registerOfflinePlayerHandlers: function() {
+      var self = this;
+
+      // Generates an id for the client
+      self.offlinePlayerClientId = Math.round(Math.random() * 10000000);
+
+      // Handle offline-player events
+      window.addEventListener("message", function(event) {
+        if(event.data.type === "offline-player-init") {
```

Comment:
@stulees @tejohnso It was added because in some tests I made I was receiving the "register.chrome.app.window" message even when running the presentation in a regular Chrome tab. I must have been doing something wrong, because I can't reproduce that anymore. Probably removing the new one is a good idea.